### PR TITLE
[TAS-374] ✨ Support check out shopping cart by card

### PR DESCRIPTION
**Pull request by WeiJun0827**
截圖 2023-09-15 下午12 10 01

---

截圖 2023-09-15 下午12 04 38

### CODE REVIEW

File: src/store/modules/nft.js [hunk 1]
```diff
@@ -395,17 +390,18 @@ const actions = {
...
     if (!classId) return undefined;
-    const { fiatPrice, listingInfo } = await this.$api.$get(
+    const { fiatPrice } = await this.$api.$get(
       api.getStripeFiatPrice({ classId })
     );
-    const data = { fiatPrice, listingInfo };
-    commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, { classId, data });
-    return data;
+    if (Array.isArray(classId)) {
+      commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, { classId, fiatPrice });
```
Comment (williamchong) [inline src/store/modules/nft.js:402]: Why check for array here but we doesn't handle it with irritated
Iterators? How about the case if it's not array?

Comment (WeiJun0827) [reply src/store/modules/nft.js:402]: It's a bit tricky that the current `fiatPrice` for a single class (not array case) includes a base tx fee, but actually, we should further break it down. Could be in this PR or another PR?

Comment (williamchong) [reply src/store/modules/nft.js:402]: Where do we display this price in this PR?

Comment (WeiJun0827) [reply src/store/modules/nft.js:402]: I was planning displaying fiat price on the button before redirecting to Stripe page.
That way we should fetch `GET /price` API when mounted.
But that will store a very long key into the store, which is not reusable, and should fetch again if any NFT class in the list is removed.

Do you think we should simply store the LIKE/USD rate and the base tx fee provided by API in store, and calculate the fiat price to display on button? Similar logic for purchasing a single class.

(Or leave it intact in this PR, abd refactor the collect modal and the shopping cart functions in some way soon?

Comment (williamchong) [reply src/store/modules/nft.js:402]: Just directly fetch the total fiat price once when we enter shopping cart page, don't calculate using like/usd

Comment (WeiJun0827) [reply src/store/modules/nft.js:402]: What about if the user removes some classes in the shopping cart?

Comment (williamchong) [reply src/store/modules/nft.js:402]: Refetch from the api

Comment (williamchong) [reply src/store/modules/nft.js:402]: I think for this we should keep it as is old behaviour, and ignore array for now
so the guard would actually be `if (!Array.isArray(classId)) {`

Comment (WeiJun0827) [reply src/store/modules/nft.js:402]: Yes, a typo

**Pull request merged**